ambry-router/src/test/java/com.github.ambry.router/PutManagerTest.java

```diff
@@ -621,6 +628,58 @@ public void testChunkFillerSleep() throws Exception {
     assertCloseCleanup();
   }
 
+  /**
+   * Tests that the replication policy in the container is respected
+   * @throws Exception
+   */
+  @Test
+  public void testReplPolicyToPartitionClassMapping() throws Exception {
+    Account refAccount = accountService.createAndAddRandomAccount();
+    Map<Container, String> containerToPartClass = new HashMap<>();
+    Iterator<Container> allContainers = refAccount.getAllContainers().iterator();
+    Container container = allContainers.next();
+    // container with null replication policy
+    container = accountService.addReplicationPolicyToContainer(container, null);
+    containerToPartClass.put(container, MockClusterMap.DEFAULT_PARTITION_CLASS);
+    container = allContainers.next();
+    // container with configured default replication policy
+    container = accountService.addReplicationPolicyToContainer(container, MockClusterMap.DEFAULT_PARTITION_CLASS);
+    containerToPartClass.put(container, MockClusterMap.DEFAULT_PARTITION_CLASS);
+    container = allContainers.next();
+    // container with a special replication policy
+    container = accountService.addReplicationPolicyToContainer(container, MockClusterMap.SPECIAL_PARTITION_CLASS);
+    containerToPartClass.put(container, MockClusterMap.SPECIAL_PARTITION_CLASS);
+    // adding this to test the random account and container case (does not actually happen if coming from the frontend)
+    containerToPartClass.put(null, MockClusterMap.DEFAULT_PARTITION_CLASS);
+
+    Map<Integer, Integer> sizeToChunkCount = new HashMap<>();
+    // simple
+    sizeToChunkCount.put(random.nextInt(chunkSize) + 1, 1);
+    int count = random.nextInt(8) + 3;
+    // composite
+    sizeToChunkCount.put(chunkSize * count, count + 1);
+    for (Map.Entry<Integer, Integer> sizeAndChunkCount : sizeToChunkCount.entrySet()) {
+      for (Map.Entry<Container, String> containerAndPartClass : containerToPartClass.entrySet()) {
+        requestAndResultsList.clear();
+        requestAndResultsList.add(new RequestAndResult(sizeAndChunkCount.getKey(), containerAndPartClass.getKey()));
+        // since the puts are processed one at a time, it is fair to check the last partition class set
+        mockClusterMap.clearLastNRequestedPartitionClasses();
+        submitPutsAndAssertSuccess(true);
+        checkLastRequestPartitionClasses(sizeAndChunkCount.getValue(), containerAndPartClass.getValue());
```

Comment:
Prefer to move the comment `// since the puts are processed one at a time, it is fair to check the last partition class set` to the position before `checkLastRequestPartitionClasses`

Reply:
Done